apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/DMEClaimTransformerV2.java

```diff
@@ -0,0 +1,444 @@
+    Timer.Context timer =
+        metricRegistry
+            .timer(MetricRegistry.name(DMEClaimTransformerV2.class.getSimpleName(), "transform"))
+            .time();
+
+    if (!(claim instanceof DMEClaim)) {
+      throw new BadCodeMonkeyException();
+    }
+    ExplanationOfBenefit eob = transformClaim((DMEClaim) claim);
+
+    timer.stop();
+    return eob;
+  }
+
+  /**
+   * @param claimGroup the CCW {@link DMEClaim} to transform
+   * @return a FHIR {@link ExplanationOfBenefit} resource that represents the specified {@link
+   *     DMEClaim}
+   */
+  private static ExplanationOfBenefit transformClaim(DMEClaim claimGroup) {
+    ExplanationOfBenefit eob = new ExplanationOfBenefit();
+
+    // Required values not directly mapped
+    eob.getMeta().addProfile(ProfileConstants.C4BB_EOB_INPATIENT_PROFILE_URL);
+
+    // Common group level fields between all claim types
+    // Claim Type + Claim ID    => ExplanationOfBenefit.id
+    // CLM_ID                   => ExplanationOfBenefit.identifier
+    // CLM_GRP_ID               => ExplanationOfBenefit.identifier
+    // BENE_ID + Coverage Type  => ExplanationOfBenefit.insurance.coverage
+    // BENE_ID                  => ExplanationOfBenefit.patient (reference)pwd
+    // FINAL_ACTION             => ExplanationOfBenefit.status
+    // CLM_FROM_DT              => ExplanationOfBenefit.billablePeriod.start
+    // CLM_THRU_DT              => ExplanationOfBenefit.billablePeriod.end
+    // CLM_PMT_AMT              => ExplanationOfBenefit.payment.amount
+    TransformerUtilsV2.mapEobCommonClaimHeaderData(
+        eob,
+        claimGroup.getClaimId(),
+        claimGroup.getBeneficiaryId(),
+        ClaimTypeV2.DME,
+        claimGroup.getClaimGroupId().toPlainString(),
+        MedicareSegment.PART_A,
+        Optional.of(claimGroup.getDateFrom()),
+        Optional.of(claimGroup.getDateThrough()),
+        Optional.of(claimGroup.getPaymentAmount()),
+        claimGroup.getFinalAction());
+
+    // map eob type codes into FHIR
+    // NCH_CLM_TYPE_CD            => ExplanationOfBenefit.type.coding
+    // EOB Type                   => ExplanationOfBenefit.type.coding
+    // Claim Type (Professional)  => ExplanationOfBenefit.type.coding
+    // NCH_NEAR_LINE_REC_IDENT_CD => ExplanationOfBenefit.extension
+    TransformerUtilsV2.mapEobType(
+        eob,
+        ClaimTypeV2.DME,
+        Optional.of(claimGroup.getNearLineRecordIdCode()),
+        Optional.of(claimGroup.getClaimTypeCode()));
+
+    // CARR_CLM_PRMRY_PYR_PD_AMT => ExplanationOfBenefit.total.amount
+    TransformerUtilsV2.addTotal(
+        eob,
+        TransformerUtilsV2.createTotalAdjudicationAmountSlice(
+            eob,
+            CcwCodebookVariable.CLM_TOT_CHRG_AMT,
+            C4BBAdjudication.PRIOR_PAYER_PAID,
+            claimGroup.getPrimaryPayerPaidAmount()));
+
+    // NCH_WKLY_PROC_DT => ExplanationOfBenefit.supportinginfo.timingDate
+    TransformerUtilsV2.addInformationWithDate(
```

Comment:
Just a note.  I changed how this is mapped on https://github.com/CMSgov/beneficiary-fhir-data/pull/488

We will want to make sure this merged first, and I'll update this under my PR.